modules/charon-core/src/test/java/org/wso2/charon3/core/utils/ResourceManagerUtilTest.java

```diff
@@ -0,0 +1,286 @@
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:addresses",
+                        "addresses", COMPLEX, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null,
+                        subSubAttributeSchemaList);
+        AttributeSchema subAttributeSchema2 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:department",
+                        "department", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema subAttributeSchema3 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:core:2.0:User:attribute3",
+                        "attribute3", STRING, true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema subAttributeSchema4 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1:attribute2",
+                        "attribute2", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, subSubAttributeSchemaList1);
+        ArrayList<AttributeSchema> subAttributeSchemaList = new ArrayList<>();
+        subAttributeSchemaList.add(subAttributeSchema1);
+        subAttributeSchemaList.add(subAttributeSchema2);
+        subAttributeSchemaList.add(subAttributeSchema3);
+        ArrayList<AttributeSchema> subAttributeSchemaList1 = new ArrayList<>();
+        subAttributeSchemaList1.add(subAttributeSchema4);
+        AttributeSchema attributeSchema1 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User",
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User", COMPLEX,
+                        false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, subAttributeSchemaList);
+        AttributeSchema attributeSchema2 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1",
+                        "attribute1", STRING,
+                        true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, subAttributeSchemaList1);
+        SCIMResourceTypeSchema scimResourceTypeSchema =
+                SCIMResourceTypeSchema.createSCIMResourceSchema(schemasList, attributeSchema1, attributeSchema2);
+
+        List<String> multiValuedAttributes = new ArrayList<>();
+        multiValuedAttributes.add
+                ("urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1:attribute2.value");
```

Comment:
A Sub attribute of a complex attribute can't be a complex attribute.
Use:
`urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1.subattribute`

Reply:
fixed in 9c935eb718ab31db04618fe9e0880afe22478953